modules/service-commons/src/main/scala/ch/epfl/bluebrain/nexus/service/commons/retrying/Retryer.scala

```diff
@@ -0,0 +1,170 @@
+package ch.epfl.bluebrain.nexus.service.commons.retrying
+
+import akka.actor.Scheduler
+import akka.pattern.after
+import ch.epfl.bluebrain.nexus.common.types.Err
+import ch.epfl.bluebrain.nexus.service.commons.retrying.Retryer.MaxRetriesException
+import ch.epfl.bluebrain.nexus.service.commons.retrying.RetryerDelays._
+import shapeless.{:+:, CNil, Coproduct, Typeable}
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Random
+
+class Retryer[C <: Coproduct](implicit C: CoproductRetryer[C], ec: ExecutionContext, s: Scheduler) {
```

Comment:
Please consider introducing Monix Task as a dependency and wrap the indexing function into a task that attempts retries based on various strategies; for example one with backoff from monix's documentation:
```
def retryBackoff[A](source: Task[A],
  maxRetries: Int, firstDelay: FiniteDuration): Task[A] = {

  source.onErrorHandleWith {
    case ex: Exception =>
      if (maxRetries > 0)
        // Recursive call, it's OK as Monix is stack-safe
        retryBackoff(source, maxRetries-1, firstDelay*2)
          .delayExecution(firstDelay)
      else
        Task.raiseError(ex)
  }
}
```